OpenMarket/OpenMarket/URLSession/URLSession.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  URLSession.swift
+//  OpenMarket
+//
+//  Created by Ayaan, junho on 2022/11/15.
+//
+
+import Foundation
+
+extension URLSession: OpenMarketURLSessionProtocol {
+    func fetchOpenMarketDataTask(query: String,
+                                 completion: @escaping (Data?, Error?) -> Void) -> URLSessionDataTask? {
+        guard let hostURL = URL(string: host),
+              let url = URL(string: query, relativeTo: hostURL) else {
+            return nil
+        }
+        
+        return self.dataTask(with: url) { (data, response, error) in
+            if let error = error {
+                completion(nil, error)
+            } else if (response as? HTTPURLResponse)?.statusCode != 200 {
+                completion(nil, OpenMarketError.badStatus)
+            } else if let data = data {
+                completion(data, nil)
+            } else {
+                completion(nil, nil)
+            }
+        }
+    }
+}
+
+extension URLSession: OpenMarketHealthFetchable { }
+extension URLSession: OpenMarketPageFetchable { }
+extension URLSession: OpenMarketProductFetchable { }
```

Comment:
어떻게보면 개인적인 차이가 있을수 있는 부분이라고 생각되는부분인데, 저는 아래와 같이 작성하고 있어요.
그 이유는 어떠한 프로토콜을 따르는지에 대해서 한번에 파악하기 더 편한다고 생각하기 때문입니다.
```suggestion
extension URLSession: OpenMarketURLSessionProtocol, OpenMarketHealthFetchable, OpenMarketPageFetchable, OpenMarketProductFetchable { 
   ...
}
```
여러분들은 어떠한 기준으로 프로토콜 채택에 관한 내용 작성하고 계시는지 궁금합니다 👀

Reply:
[4cd27d0](https://github.com/yagom-academy/ios-open-market/pull/206/commits/4cd27d0be0a9742095b407a8f85358c0b2006348)
프로토콜을 채택하고 기능 구현이 필요할 때에는 저희가 작성한 코드처럼 작성하면, 채택한 프로토콜의 기능만 구분해서 볼 수 있기 때문에 가독성 면에서 더 좋은 것 같습니다.
하지만 현재 프로토콜에서 default로 메서드를 구현해놓고 해당 메서드를 그대로 사용하기 때문에 태태가 제안해 주신 코드가 가독성 면에서 더 좋은 것 같아 수정했습니다!